src/frontend/screens/Settings/ProjectSettings/DeviceName/index.tsx

```diff
@@ -0,0 +1,223 @@
+    <ScrollView contentContainerStyle={styles.container}>
+      {data?.name &&
+        (mode === 'display' ? (
+          <DisplayMode deviceName={data.name} navigation={navigation} />
+        ) : (
+          <EditMode deviceName={data.name} navigation={navigation} />
+        ))}
+    </ScrollView>
+  );
+};
+
+function DisplayMode({
+  deviceName,
+  navigation,
+}: {
+  deviceName: string;
+  navigation: React.ComponentProps<
+    NativeNavigationComponent<'DeviceName'>
+  >['navigation'];
+}) {
+  const {formatMessage: t} = useIntl();
+
+  React.useEffect(
+    function updateNavigationOptions() {
+      navigation.setOptions({
+        headerRight: () => (
+          <IconButton onPress={() => navigation.setParams({mode: 'edit'})}>
+            <EditIcon />
+          </IconButton>
+        ),
+      });
+    },
+    [navigation],
+  );
+
+  return (
+    <FieldRow label={t(m.deviceNameDisplayHeader)}>
+      <Text>{deviceName}</Text>
+    </FieldRow>
+  );
+}
+
+DeviceName.navTitle = m.title;
+
+function EditMode({
+  deviceName,
+  navigation,
+}: {
+  deviceName: string;
+  navigation: React.ComponentProps<
+    NativeNavigationComponent<'DeviceName'>
+  >['navigation'];
+}) {
+  const {formatMessage: t} = useIntl();
+  const {mutate: updateDeviceName} = useEditDeviceInfo();
+
+  const {control, getValues, handleSubmit} = useForm<{
+    deviceName: string;
+  }>({
+    defaultValues: {deviceName},
+  });
+
+  React.useEffect(
+    function showDiscardChangesAlert() {
+      const unsubscribe = navigation.addListener('beforeRemove', event => {
+        event.preventDefault();
+
+        const nameHasChanges = getValues('deviceName') !== deviceName;
+
+        if (nameHasChanges) {
+          Alert.alert(t(m.discardChangesAlertTitle), undefined, [
+            {style: 'cancel', text: t(m.discardChangesAlertCancelText)},
+            {
+              style: 'default',
+              text: t(m.discardChangesAlertConfirmText),
+              onPress: () => {
+                // TODO: Is going to display mode desired behavior here? Or should it go back to previous screen
+                navigation.setParams({mode: 'display'});
+              },
+            },
+          ]);
+        } else {
+          // TODO: Is going to display mode desired behavior here? Or should it go back to previous screen
+          navigation.setParams({mode: 'display'});
+        }
+      });
+
+      return () => {
+        unsubscribe();
+      };
+    },
+    [t, navigation, getValues, deviceName],
+  );
+
+  React.useEffect(
+    function updateNavigationOptions() {
+      navigation.setOptions({
+        headerRight: () => {
+          return (
+            <IconButton
+              onPress={handleSubmit(value => {
+                updateDeviceName(value.deviceName, {
+                  onSuccess: () => navigation.setParams({mode: 'display'}),
+                  onError: _err => {
+                    // TODO: Handle errors
+                  },
```

Comment:
highlighting just in case i need to do something here. my guess is that this could be a follow-up that implements an error modal pattern